lightning-block-sync/src/http.rs

```diff
@@ -636,7 +636,7 @@ pub(crate) mod client_tests {
 	#[test]
 	fn connect_to_unresolvable_host() {
 		match HttpClient::connect(("example.invalid", 80)) {
-			Err(e) => assert_eq!(e.kind(), std::io::ErrorKind::Other),
+			Err(_) => { /* rust libstd currently doesn't have a specific error for Host Not Found for us to match against */ },
```

Comment:
Thoughts on this patch or similar:

```
diff --git a/lightning-block-sync/src/http.rs b/lightning-block-sync/src/http.rs
index 1f30bea1..93f7bafb 100644
--- a/lightning-block-sync/src/http.rs
+++ b/lightning-block-sync/src/http.rs
@@ -636,7 +636,9 @@ pub(crate) mod client_tests {
        #[test]
        fn connect_to_unresolvable_host() {
                match HttpClient::connect(("example.invalid", 80)) {
-                       Err(_) => { /* rust libstd currently doesn't have a specific error for Host Not Found for us to match against */ },
+                       Err(e) => {
+                               assert_eq!(e.to_string(), "failed to lookup address information: Name or service not known");
+                       },
                        Ok(_) => panic!("Expected error"),
                }
        }
```

Reply:
Good point. I went with `e.to_string().contains("Name or service not known")` as I'm not 100% sure `to_string` is guaranteed to never change, either.

Reply:
Turns out its not cross-platform. I think `failed to lookup address information` will be, for now, though.